certbot-nginx/certbot_nginx/parser.py

```diff
@@ -344,6 +312,16 @@ def add_server_directives(self, vhost, directives, replace):
         except errors.MisconfigurationError as err:
             raise errors.MisconfigurationError("Problem in %s: %s" % (filename, str(err)))
 
+def _parse_flat_file(ssl_options):
```

Comment:
given that was already have a complicated general mechanism for parsing files, I'm going to return this to its original name and leave it as a special-use function.

If #583 ever gets resolved, that would be a good time to fix our general machinery for parsing files and included files and figuring out which directives apply where.